src/timeseriesflattener/flattened_dataset.py

```diff
@@ -526,196 +615,174 @@ def _add_incident_outcome(
+        Also add patient's birth date. Has its own function because of its very frequent use.
 
         Args:
-            direction (str): Whether to look ahead or behind.
-            interval_days (float): How far to look
-            df (DataFrame): Source dataframe
-            timestamp_pred_colname (str): Name of timestamp column for predictions in df.
-            timestamp_value_colname (str): Name of timestamp column for values in df.
+            id2date_of_birth (DataFrame): Two columns, id and date_of_birth.
+            input_date_of_birth_col_name (str, optional): Name of the date_of_birth column in id2date_of_birth.
+                Defaults to "date_of_birth".
+            output_prefix (str, optional): Prefix for the output column. Defaults to "pred".
+            birth_year_as_predictor (bool, optional): Whether to add birth year as a predictor. Defaults to False.
+        """
+        if id2date_of_birth[input_date_of_birth_col_name].dtype != "<M8[ns]":
+            try:
+                id2date_of_birth[input_date_of_birth_col_name] = pd.to_datetime(
+                    id2date_of_birth[input_date_of_birth_col_name],
+                    format="%Y-%m-%d",
+                )
+            except ValueError as e:
+                raise ValueError(
+                    f"Conversion of {input_date_of_birth_col_name} to datetime failed, doesn't match format %Y-%m-%d. Recommend converting to datetime before adding.",
+                ) from e
 
-        Raises:
-            ValueError: If direction is niether ahead nor behind.
+        output_age_col_name = f"{output_prefix}_age_in_years"
 
-        Returns:
-            DataFrame
-        """
-        df["time_from_pred_to_val_in_days"] = (
-            (df[timestamp_value_colname] - df[timestamp_pred_colname])
-            / (np.timedelta64(1, "s"))
-            / 86_400
+        self._add_static_info(
+            static_spec=AnySpec(
+                values_df=id2date_of_birth,
+                input_col_name_override=input_date_of_birth_col_name,
+                prefix=output_prefix,
+                # We typically don't want to use date of birth as a predictor,
+                # but might want to use transformations - e.g. "year of birth" or "age at prediction time".
+                feature_name=input_date_of_birth_col_name,
+            ),
         )
-        # Divide by 86.400 seconds/day
 
-        if direction == "ahead":
-            df["is_in_interval"] = (
-                df["time_from_pred_to_val_in_days"] <= interval_days
-            ) & (df["time_from_pred_to_val_in_days"] > 0)
-        elif direction == "behind":
-            df["is_in_interval"] = (
-                df["time_from_pred_to_val_in_days"] >= -interval_days
-            ) & (df["time_from_pred_to_val_in_days"] < 0)
-        else:
-            raise ValueError("direction can only be 'ahead' or 'behind'")
+        data_of_birth_col_name = f"{output_prefix}_{input_date_of_birth_col_name}"
```

Comment:
wait, so we also return the date of birth as something like `pred_date_of_birth` as well as e.g. `pred_age`?
I suggest dropping it and only keeping age (unless `birth_year_as_predictor` is True) as we should never have datetimes as predictors

Reply:
also, f2 to rename to `date_of_birth_col_name`